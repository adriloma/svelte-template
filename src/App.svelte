<script>
import {onMount} from 'svelte';

let selected = '';
let pokemonList;
let activePokemon;
onMount (async () => {
	const response = await fetch('https://pokeapi.co/api/v2/pokemon?limit=100&offset=200');
	pokemonList = await response.json();
})

async function loadPokemon() {
	debugger;
	const pokeToLoad = pokemonList.results[selected];
	const response = await fetch(pokeToLoad.url);
	activePokemon = await response.json();
}

</script>

<main>

</main>

<style>
	main {
		text-align: center;
		padding: 1em;
		max-width: 100%;
		margin: 0 auto;
	& h1 {
			color: #ff3e00;
			text-transform: uppercase;
			font-size: 4em;
			font-weight: 100;
			color: red;
		}
	}
</style>
<main>
	<h1>Pokemon slider</h1>
	{#if pokemonList}
	<h3>{selected || 'choose one!'}</h3>
	<input type="range" min=0 max={pokemonList.results.length - 1} bind:value={selected} on:change={loadPokemon}>
	{:else}
	loading...
	{/if}
	{#if activePokemon}
	<img src={activePokemon.sprites.front_default} alt={activePokemon.name}/>
	{/if}
</main>