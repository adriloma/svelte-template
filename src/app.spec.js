import App from './App.svelte';
import { render, fireEvent } from '@testing-library/svelte';
import '@testing-library/jest-dom';



it('renders', async () => {
    const { container } = render(App);
    expect(container).toMatchSnapshot();
});

it('works', async () => {
    const { getByText, getByTestId } = render(App);
    const increment = getByText('+');
    const decrement = getByText('-');
    const reset = getByText('Reset');
    const counter = getByTestId('counter');

    await fireEvent.click(increment);
    // with jest
    expect(counter.textContent).toBe('1');
    // with jest-dom
    expect(counter).toHaveTextContent('1');
});