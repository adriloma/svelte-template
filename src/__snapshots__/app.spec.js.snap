// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<body>
  <div>
    <main>
      <h1>
        Hello 
        Adri√°n
        !
      </h1>
       
      <p>
        Visit the 
        <a
          href="https://svelte.dev/tutorial"
        >
          Svelte tutorial
        </a>
         to learn how to build Svelte apps.
      </p>
       
      <p>
        You can play with the counter: 
        <span
          data-testid="counter"
        >
          0
        </span>
      </p>
       
      <button>
        +
      </button>
       
      <button>
        -
      </button>
       
      <button>
        Reset
      </button>
    </main>
  </div>
</body>
`;
